cub/cub/device/dispatch/dispatch_reduce.cuh

```diff
@@ -363,7 +363,7 @@ __global__ void DeviceSegmentedReduceKernel(
     OutputIteratorT d_out,
     BeginOffsetIteratorT d_begin_offsets,
     EndOffsetIteratorT d_end_offsets,
-    int /*num_segments*/,
+    OffsetT /*num_segments*/,
```

Comment:
Oh, I see! My main motivation was being able to handle 64-bit indices. I saw that in the dispatch portion, `num_segments` is constrained to be of type `OffsetT`. Since it is not possible to handle 64-bit `num_segments`, then your solution seems like the way to go. I will undo the `NumSegmentsT` part and leave in the type deduction for `OffsetT`.

Reply:
The technical specifications say that the maximum x-dimension of `blockDim` is 2^31-1, but the maximum y- and z-dimensions are 65535 (2^16-1). Could we maybe use the y- and z-dimensions to increase the theoretical maximum to 2^63-1?